### ISSUE TITLE
macOS Sierra Support

### ISSUE THREAD

**Issue by lmurgatroyd** [JamesFator/BTHSControl#12]
It appears that BTHSControl no longer functions with macOS Sierra, as tested with the GM released on September 8th.

**Comment by ev0rtex** [comment]
I am planning to upgrade to Sierra soon...perhaps I need to hold off a bit until this is resolved

**Comment by dmzimmerman** [comment]
Indeed, I just built it today (on the latest dev release of Sierra), and it does not work. Lots of errors like this in the console:

default 16:22:20.441531 -0700   AVRCPAgent  [establishKernelConnection] Received an error from IOServiceOpen() - 0xe00002c7.  NULLing out io_service_t.

I'm guessing something needs to be updated for new Sierra kernel stuff.

**Comment by ev0rtex** [comment]
~~Is there any chance of getting this fixed? I'm not experienced with any form of Mac OS development or I'd attempt to figure this out myself.~~

**EDIT:** Sorry, I just completely uninstalled and rebuilt/installed again and it is working for me on Sierra currently.

**Comment by JamesFator** [comment]
Unfortunately Sierra is not on my roadmap at the moment. I've swapped over to Linux for work and have had no free time to tackle this. If anyone does find a fix, or things end up working like ev0rtex suggests, let me know. Other than that, I might potentially have time to look at it when the holidays roll around.

**Comment by wmertens** [comment]
I gave it a shot on 10.12.2 and this gets into the logs:

```
default	11:45:08.790896 +0100	AVRCPAgent	BTHSControlPlugin installed
default	11:45:08.792618 +0100	AVRCPAgent	Setting Delegate: <BluetoothAVRCPAgent: 0x7f94f6706c90>
default	11:45:08.799402 +0100	AVRCPAgent	[establishKernelConnection] Received an error from IOServiceOpen() - 0xe00002c7.  NULLing out io_service_t.
```

Not working.

**Comment by gil** [comment]
Since no one mentioned and there's no plan for an update, I found this alternative to BTHSControl. The beta version is working here for Sierra -> https://github.com/jguice/mac-bt-headset-fix#beta-version
